app/models/comment.rb

```diff
@@ -64,6 +75,12 @@ def initialize(person, target, text)
       super(person, target)
     end
 
+    def build(*args)
+      super.tap do |comment|
+        comment.filter_mentions(comment.people_allowed_to_be_mentioned)
```

Comment:
So [looks like](https://travis-ci.org/diaspora/diaspora/jobs/147856327#L653) mentioning in status message and post via federation may bypass some visibility restrictions.